main.tf

```
@@ -1,16 +1,149 @@
+  name_prefix = "${var.name}-"
+  description = "Role for ${var.name} to create secret"
+
+  assume_role_policy = data.aws_iam_policy_document.assume.json
+  tags               = var.tags
+
+  lifecycle {
+    create_before_destroy = true
+  }
 }
 
 locals {
-  # tflint-ignore: terraform_unused_declarations
-  name = module.tags.name
-  # tflint-ignore: terraform_unused_declarations
-  tags = module.tags.tags_no_name
+  zipfile = "${path.module}/lambda/build/lambda-create-password-secret-production.zip"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_basic_execution" {
+  role       = aws_iam_role.this.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role_policy_attachment" "secretsmanager_read_write" {
+  role       = aws_iam_role.this.name
+  policy_arn = "arn:aws:iam::aws:policy/SecretsManagerReadWrite"
+}
+
+resource "aws_lambda_function" "this" {
+  depends_on = [
+    module.lambda_package
+  ]
+
+  filename         = local.zipfile
+  function_name    = var.name
+  handler          = "index.handler"
+  role             = aws_iam_role.this.arn
+  runtime          = "nodejs12.x"
+  source_code_hash = try(filebase64sha256(local.zipfile), null)
+
+  lifecycle {
+    ignore_changes = [
+      filename,
+      last_modified
+    ]
+  }
+
+  tags = merge(
+    var.tags,
+    {
+      "Name" = var.name
+    }
+  )
+}
+
+resource "aws_secretsmanager_secret" "this" {
+  name_prefix = "${var.name}-"
+  description = var.secret_description
+  tags = merge(
+    var.tags,
+    {
+      Name = var.name
+    }
+  )
+}
+
+locals {
+  password_params = {
+    length             = var.length
+    upper              = var.upper
+    min_upper          = var.min_upper
+    lower              = var.lower
+    min_lower          = var.min_lower
+    number             = var.number
+    min_numeric        = var.min_numeric
+    special            = var.special
+    min_special        = var.min_special
+    override_special   = var.override_special
+    secret_name        = aws_secretsmanager_secret.this.name
+    secret_description = var.secret_description
+  }
+}
+
+# Using a random_string as a trigger for when the password should be re-created.
+# This allows the Terraform re-creation logic to be duplicated exactly.
+resource "random_string" "trigger" {
+  keepers = var.keepers
+
+  length           = var.length
+  upper            = var.upper
+  min_upper        = var.min_upper
+  lower            = var.lower
+  min_lower        = var.min_lower
+  number           = var.number
+  min_numeric      = var.min_numeric
+  special          = var.special
+  min_special      = var.min_special
+  override_special = var.override_special
+}
+
+module "lambda_invocation" {
+  source  = "matti/resource/shell"
+  version = "~>1.0.7"
+
+  command = "aws lambda invoke --function-name ${aws_lambda_function.this.function_name} --payload '${jsonencode(local.password_params)}' /tmp/lambda_invocation_output"
```

```suggestion
  command = "aws lambda invoke --function-name ${aws_lambda_function.this.function_name} --payload '${jsonencode(local.password_params)}' > /tmp/lambda_invocation_output"
```

Assuming you need a `>` here?

---

Nope. the `aws_lambda_invoke` function expects a file path for the output of the function itself, and stdout is reserved for the info about the return code from the function.

I don't like it, but that's how it works.